siriuspy/siriuspy/devices/egun.py

```diff
@@ -132,6 +135,14 @@ def is_on(self):
         """."""
         return self['swstatus'] == self.PWRSTATE.On
 
+    def wait_current(self, value, tol=0.20, timeout=DEF_TIMEOUT):
+        """Wait current to reach value with tolerance 'tol'."""
+        _t0 = _time.time()
+        while _time.time() - _t0 < timeout:
+            if abs(self.current - value) < tol:
+                return True
```

Comment:
@anacso17 , don't you think we should insert a `time.sleep` in the loop in order to ease on the cpu ?

Reply:
thanks!!